public/src/components/channelManagement/richTextEditor.tsx

```diff
@@ -0,0 +1,407 @@
+}
+
+// ReMirror/ProseMirror FLOATING LINK MENU BUTTONS
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+const FloatingLinkToolbar = () => {
+  const {
+    isEditing,
+    linkPositioner,
+    clickEdit,
+    onRemove,
+    submitHref,
+    href,
+    setHref,
+    cancelHref,
+  } = useFloatingLinkState();
+  const active = useActive();
+  const activeLink = active.link();
+  const { empty } = useCurrentSelection();
+  const linkEditItems: ToolbarItemUnion[] = useMemo(
+    () => [
+      {
+        type: ComponentItem.ToolbarGroup,
+        label: 'Link',
+        items: activeLink
+          ? [
+              // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+              { type: ComponentItem.ToolbarButton, onClick: () => clickEdit(), icon: 'pencilLine' },
+              { type: ComponentItem.ToolbarButton, onClick: onRemove, icon: 'linkUnlink' },
+            ]
+          : [
+              // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+              { type: ComponentItem.ToolbarButton, onClick: () => clickEdit(), icon: 'link' },
+            ],
+      },
+    ],
+    [clickEdit, onRemove, activeLink],
+  );
+
+  const items: ToolbarItemUnion[] = useMemo(() => linkEditItems, [linkEditItems]);
+
+  return (
+    <>
+      <FloatingToolbar items={items} positioner="selection" placement="top" enabled={!isEditing} />
+      <FloatingToolbar
+        items={linkEditItems}
+        positioner={linkPositioner}
+        placement="bottom"
+        enabled={!isEditing && empty}
+      />
+      <FloatingWrapper
+        positioner="always"
+        placement="bottom"
+        enabled={isEditing}
+        renderOutsideEditor
+      >
+        <input
+          style={{ zIndex: 20 }}
+          autoFocus
+          placeholder="Enter link..."
+          // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+          onChange={event => setHref(event.target.value)}
+          value={href}
+          // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+          onKeyDown={event => {
+            const { key } = event;
+            if (key === 'Enter') {
+              submitHref();
+            }
+            if (key === 'Escape') {
+              cancelHref();
+            }
+          }}
+        />
+      </FloatingWrapper>
+    </>
+  );
+};
+
+// ReMirror/ProseMirror TOP MENU BUTTONS
+const RichTextMenu: React.FC<RichTextMenuProps> = ({
+  classes,
+  disabled,
+  label,
+}: RichTextMenuProps) => {
+  const chain = useChainedCommands();
+  const active = useActive();
+
+  return (
+    <div>
+      <span className={classes.fieldLabel}>{label != null ? label : 'Editable field'}</span>
+      {!disabled && (
+        <>
+          <button
+            className={active.bold() ? 'remirror-button-active' : 'remirror-button'}
+            // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```

Comment:
the linter doesn't appear to care about these (`npm run lint`) - can they be removed?

Reply:
I'll see if they can be removed without linter complaints

Reply:
11 warning lines removed, linter not complaining.